### PR TITLE
FB8-80: Display row information in processlist for rbr threads

### PR DESCRIPTION
**Pull request by inikep**
Jira ticket: https://jira.percona.com/browse/FB8-80

Reference Patch: https://github.com/facebook/mysql-5.6/commit/564824a

Set row_query using the fields unpacked from the row event. Use row_query in show processlist.

Test Plan: mtr

Originally Reviewed By: tianx

### CODE REVIEW

File: sql/rpl_record.cc [hunk 1]
```diff
@@ -400,6 +400,36 @@ size_t pack_row(TABLE *table, MY_BITMAP const *columns_in_image,
   DBUG_RETURN(static_cast<size_t>(pack_ptr - row_data));
 }
 
+static void insert_row_fields(std::string *row_query, TABLE *table) {
+  static const size_t max_field_len = 100;
```
Comment (percona-ysorokin) [inline sql/rpl_record.cc]:
```suggestion
  static constexpr size_t max_field_len = 100;
```

File: sql/rpl_record.cc [hunk 2]
```diff
@@ -400,6 +400,36 @@ size_t pack_row(TABLE *table, MY_BITMAP const *columns_in_image,
   DBUG_RETURN(static_cast<size_t>(pack_ptr - row_data));
 }
 
+static void insert_row_fields(std::string *row_query, TABLE *table) {
```
Comment (percona-ysorokin) [inline sql/rpl_record.cc]: There is no checking for `row_query != nullptr`.
Pass inout parameters by reference
```suggestion
static void insert_row_fields(std::string &row_query, TABLE *table) {
```

File: sql/rpl_record.cc [hunk 3]
```diff
@@ -457,7 +487,8 @@ bool unpack_row_with_column_info(TABLE *table, uchar const *const row_data,
                                  MY_BITMAP const *column_image,
                                  uchar const **const row_image_end_p,
                                  uchar const *const event_end, bool only_seek,
-                                 table_def *tabledef, TABLE *conv_table) {
+                                 table_def *tabledef, TABLE *conv_table,
+                                 std::string *row_query) {
```
Comment (percona-ysorokin) [inline sql/rpl_record.cc]: pass by reference

File: sql/rpl_record.cc [hunk 4]
```diff
@@ -657,7 +689,8 @@ bool unpack_row(Relay_log_info const *rli, TABLE *table,
                 uchar const **const row_image_end_p,
                 uchar const *const event_end,
                 enum_row_image_type row_image_type,
-                bool event_has_value_options, bool only_seek) {
+                bool event_has_value_options, bool only_seek,
+                std::string *row_query) {
```
Comment (percona-ysorokin) [inline sql/rpl_record.cc]: pass by reference

File: sql/rpl_record.h [hunk 5]
```diff
@@ -46,7 +47,8 @@ bool unpack_row(Relay_log_info const *rli, TABLE *table,
                 uchar const **const row_image_end_p,
                 uchar const *const event_end,
                 enum_row_image_type row_image_type,
-                bool event_has_value_options, bool only_seek);
+                bool event_has_value_options, bool only_seek,
+                std::string *row_query = nullptr);
```
Comment (percona-ysorokin) [inline sql/rpl_record.h]: pass by reference, no default param

File: sql/sql_show.cc [hunk 6]
```diff
@@ -2004,8 +2004,17 @@ class List_process_list : public Do_THD_Impl {
     /* INFO */
     mysql_mutex_lock(&inspect_thd->LOCK_thd_query);
     {
-      const char *query_str = inspect_thd->query().str;
-      size_t query_length = inspect_thd->query().length;
+      const char *query_str;
+      size_t query_length;
+
+      if (inspect_thd->row_query.size()) {
```
Comment (percona-ysorokin) [inline sql/sql_show.cc]:
```suggestion
      if (!inspect_thd->row_query.empty()) {
```

File: sql/sql_show.cc [hunk 7]
```diff
@@ -2191,8 +2200,17 @@ class Fill_process_list : public Do_THD_Impl {
...
```
Comment (percona-ysorokin) [inline sql/sql_show.cc]:
```suggestion
      if (!inspect_thd->row_query.empty()) {
```

**Review by percona-ysorokin — APPROVED** [review]
LGTM
(just minor stylistic comments)

**Review by hermanlee — CHANGES_REQUESTED** [review]
The changes look good, but the follow tests look to be failing:

rpl.rpl_multi_source_row_stages
rpl.rpl_invalid_mem_access_with_rows_query_log_event

**Comment by inikep** [comment]
Failing tests were fixed and the commit was rebased.

**Review by hermanlee — APPROVED** [review]

**Pull request closed without merging**